src/common/compute/backends/sciserver-compute/Client.js

```diff
@@ -220,30 +220,32 @@ define([
         async _getFile (jobInfo, filename) {
             const state = await this.getJobState(jobInfo);
             if (state) {
-                const baseUrl = 'https://apps.sciserver.org/fileservice/api/file/';
                 const filepath = state.resultsFolderURI.replace(/\/?$/, '/') + filename;
-                const fileUrl = baseUrl + this._getEncodedFilePath(filepath);
-
-                const response = await this.fetch(fileUrl);
-                return await response.text();
+                const {config, dataInfo} = this._getStorageConfigAndDataInfo(filepath);
+                const storage = await Storage.getClient('sciserver-files', this.logger, config);
+                return (await storage.getFile(dataInfo)).toString('utf-8');
             }
         }
 
         async _deleteFileDir (state) {
-            const baseUrl = 'https://apps.sciserver.org/fileservice/api/data/';
             const filepath = state.command.split(' ').pop();
-            const fileUrl = baseUrl + this._getEncodedFilePath(filepath);
-            const response = await this.fetch(fileUrl, {method: 'DELETE'});
-            return await response.text();
+            const {dataInfo, config} = this._getStorageConfigAndDataInfo(filepath);
+            const storage = await Storage.getClient('sciserver-files', this.logger, config);
+            await storage.deleteDir(dataInfo.data.filename);
         }
 
-        _getEncodedFilePath (filepath) {
+        _getStorageConfigAndDataInfo (filepath) {
             const dirs = filepath.split('/').slice(4);
-            const filename = dirs.pop();
-            const drive = dirs.slice(0, 3).join('/') + '/';
-            const dirname = '/' + dirs.slice(3).join('/');
-
-            return drive + encodeURIComponent(dirname) + '/' + filename;
+            let [volumePool, username, volume] = dirs.slice(0, 3);
```

Comment:
As a minor point, it would be better to use `this.username` for the username and maybe rename this variable to something like `owner`. It shouldn't really increase the complexity but would make this code work for shared directories, too.

Reply:
Yeah, I agree. This would be helpful especially (if we are to store Job results in a shared directory).